### PR TITLE
Support for editing single exercise deadlines

### PR DESCRIPTION
**Pull request by jsopakar**
Some tests might fail on this, they are fixed on different branch and will be on another PR. Test concerning these changes should work.

### CODE REVIEW

**Comment by jsopakar** [comment]
Test fixes should be actually on PR #288

File: app/assets/javascripts/courses.js [hunk 1]
```diff
@@ -20,4 +20,75 @@ $(document).ready(function() {
...
     $(".feedback-show-button").click(toggle_feedback_reply);
     $(".feedback-hide-button").click(toggle_feedback_reply);
+
+    /*
+     * Deadline group toggle functions
+     */
+
+    var deadline_group_toggle_statuses = {};
+
+    var toggle_deadline_group_status = function (id) {
```
Comment (jamox) [inline app/assets/javascripts/courses.js]: Rm space between function and left bracket.

File: app/assets/javascripts/courses.js [hunk 2]
```diff
@@ -20,4 +20,75 @@ $(document).ready(function() {
...
+    var get_deadline_group_toggle_status = function (id) {
+        if (deadline_group_toggle_statuses.hasOwnProperty(id)) {
+            return deadline_group_toggle_statuses[id];
+        } else {
+            return false;
+        }
+    };
+
+    var toggle_deadline_grouping = function (e) {
+        var id = e.target.id.substring(8);
```
Comment (jamox) [inline app/assets/javascripts/courses.js]: this seems like a hack

File: app/assets/stylesheets/courses.css [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    margin: 25px 8px 0 0;
+}
+
+.unlock-deadline-field {
+    display: none;
+}
+
+.single-exercise-deadlines {
+    display:none;
+    background-color: #F9F9F9
```
Comment (jamox) [inline app/assets/stylesheets/courses.css]: missins ;

File: app/models/course.rb [hunk 4]
```diff
@@ -313,6 +313,13 @@ def material_url=(material)
     super(material)
   end
 
+  def contains_unlock_deadlines?
+    exercise_groups.each do |group|
```
Comment (jamox) [inline app/models/course.rb]: .any? with almost same block maybe

File: app/models/exercise_group.rb [hunk 5]
```diff
@@ -62,6 +62,20 @@ def soft_group_deadline
...
 
+  # Returns true if all exercises in this group have the same deadline
+  def uniform_group_deadlines?
+    exercises(false).map do |e|
+      [e.static_deadline, e.unlock_deadline, e.soft_static_deadline, e.soft_unlock_deadline]
+    end.uniq.length == 1
+  end
+
+  def contains_unlock_deadlines?
+    exercises(false).each do |e|
```
Comment (jamox) [inline app/models/exercise_group.rb]: .any? maybe

**Pull request merged**